packages/components/MazPhoneNumberInput/_main.vue

```diff
@@ -308,55 +305,60 @@ export default {
-      return {
-        countryCode,
-        isValid: false,
-        ...(phoneNumber && (phoneNumber !== '')
-          ? { phoneNumber : phoneNumber }
-          : null
-        ),
-        ...(parsing
-          ? {
-            countryCode: parsing.country || countryCode,
-            countryCallingCode: parsing.countryCallingCode,
-            formattedNumber: parsing.number,
-            nationalNumber: parsing.nationalNumber,
-            isValid: parsing.isValid(),
-            type: parsing.getType(),
-            formatInternational: parsing.formatInternational(),
-            formatNational: parsing.formatNational(),
-            uri: parsing.getURI(),
-            e164: parsing.format('E.164')
-          }
-          : null
-        )
+    async getParsePhoneNumberFromString ({ phoneNumber, countryCode }) {
+      try {
+        const parsing = phoneNumber ? parsePhoneNumberFromString(phoneNumber, countryCode) : null
+        this.results = {
+          countryCode,
+          isValid: false,
+          ...(phoneNumber && (phoneNumber !== '')
+            ? { phoneNumber : phoneNumber }
+            : null
+          ),
+          ...(parsing
+            ? {
+              countryCode: parsing.country || countryCode,
+              countryCallingCode: parsing.countryCallingCode,
+              asYouTypeNumber: parsing.number,
+              nationalNumber: parsing.nationalNumber,
+              isValid: parsing.isValid(),
+              type: parsing.getType(),
+              formatInternational: parsing.formatInternational(),
+              formatNational: parsing.formatNational(),
+              uri: parsing.getURI(),
+              e164: parsing.format('E.164')
+            }
+            : null
+          )
+        }
+      } catch (e) {
+        return new Error('MazPhoneNumberInput: Impossible to parse phone number')
       }
     },
-    async emitValues (payload) {
-      const { inputValue, lastKeyPressed, getParsePhoneNumberFromString } = this
-
+    async buildResults (payload) {
+      const { asYouTypeNumber, lastKeyPressed, getParsePhoneNumberFromString, getAsYouTypeFormat } = this
       const backSpacePressed = lastKeyPressed === 8
 
       await this.$nextTick()
-      const lastCharacOfPhoneNumber = inputValue ? inputValue.trim().slice(-1) : false
+
+      const lastCharacOfPhoneNumber = asYouTypeNumber ? asYouTypeNumber.slice(asYouTypeNumber.length - 1) : false
       if (backSpacePressed && lastCharacOfPhoneNumber && (lastCharacOfPhoneNumber === ')')) {
-        return this.inputValue = inputValue.slice(0, -2)
+        this.asYouTypeNumber = asYouTypeNumber.slice(0, -1)
       }
 
-      this.results = getParsePhoneNumberFromString(payload)
+      await getParsePhoneNumberFromString(payload)
+
+      await getAsYouTypeFormat({
+        ...payload,
+        phoneNumber: this.results.formatNational || payload.phoneNumber
```

Comment:
For RU. If remove `this.results.formatNational || ` work correctly. But isValid work bad